In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

## Data set road

In [2]:
pwd

'/Users/kim-wontae/Desktop/deschool/data/house-prices-advanced-regression-techniques'

In [3]:
train = pd.read_csv("/Users/kim-wontae/Desktop/deschool/data/house-prices-advanced-regression-techniques/train.csv")
print(train.shape)
print(train.describe())
train.head()

(1460, 81)
                Id   MSSubClass  LotFrontage        LotArea  OverallQual  \
count  1460.000000  1460.000000  1201.000000    1460.000000  1460.000000   
mean    730.500000    56.897260    70.049958   10516.828082     6.099315   
std     421.610009    42.300571    24.284752    9981.264932     1.382997   
min       1.000000    20.000000    21.000000    1300.000000     1.000000   
25%     365.750000    20.000000    59.000000    7553.500000     5.000000   
50%     730.500000    50.000000    69.000000    9478.500000     6.000000   
75%    1095.250000    70.000000    80.000000   11601.500000     7.000000   
max    1460.000000   190.000000   313.000000  215245.000000    10.000000   

       OverallCond    YearBuilt  YearRemodAdd   MasVnrArea   BsmtFinSF1  ...  \
count  1460.000000  1460.000000   1460.000000  1452.000000  1460.000000  ...   
mean      5.575342  1971.267808   1984.865753   103.685262   443.639726  ...   
std       1.112799    30.202904     20.645407   181.066207   456

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0   1          60       RL         65.0     8450   Pave   NaN      Reg   
1   2          20       RL         80.0     9600   Pave   NaN      Reg   
2   3          60       RL         68.0    11250   Pave   NaN      IR1   
3   4          70       RL         60.0     9550   Pave   NaN      IR1   
4   5          60       RL         84.0    14260   Pave   NaN      IR1   

  LandContour Utilities  ... PoolArea PoolQC Fence MiscFeature MiscVal MoSold  \
0         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
1         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      5   
2         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      9   
3         Lvl    AllPub  ...        0    NaN   NaN         NaN       0      2   
4         Lvl    AllPub  ...        0    NaN   NaN         NaN       0     12   

  YrSold  SaleType  SaleCondition  SalePrice  
0   2008        WD         Normal     208500  
1   2007        WD         Normal     181500  
2   2008        WD         Normal     223500  
3   2006        WD        Abnorml     140000  
4   2008        WD         Normal     250000  

[5 rows x 81 columns]

In [4]:
test = pd.read_csv("/Users/kim-wontae/Desktop/deschool/data/house-prices-advanced-regression-techniques/test.csv")
print(test.shape)
print(test.describe())
test.head()

(1459, 80)
                Id   MSSubClass  LotFrontage       LotArea  OverallQual  \
count  1459.000000  1459.000000  1232.000000   1459.000000  1459.000000   
mean   2190.000000    57.378341    68.580357   9819.161069     6.078821   
std     421.321334    42.746880    22.376841   4955.517327     1.436812   
min    1461.000000    20.000000    21.000000   1470.000000     1.000000   
25%    1825.500000    20.000000    58.000000   7391.000000     5.000000   
50%    2190.000000    50.000000    67.000000   9399.000000     6.000000   
75%    2554.500000    70.000000    80.000000  11517.500000     7.000000   
max    2919.000000   190.000000   200.000000  56600.000000    10.000000   

       OverallCond    YearBuilt  YearRemodAdd   MasVnrArea   BsmtFinSF1  ...  \
count  1459.000000  1459.000000   1459.000000  1444.000000  1458.000000  ...   
mean      5.553804  1971.357779   1983.662783   100.709141   439.203704  ...   
std       1.113740    30.390071     21.130467   177.625900   455.268042  

Id  MSSubClass MSZoning  LotFrontage  LotArea Street Alley LotShape  \
0  1461          20       RH         80.0    11622   Pave   NaN      Reg   
1  1462          20       RL         81.0    14267   Pave   NaN      IR1   
2  1463          60       RL         74.0    13830   Pave   NaN      IR1   
3  1464          60       RL         78.0     9978   Pave   NaN      IR1   
4  1465         120       RL         43.0     5005   Pave   NaN      IR1   

  LandContour Utilities  ... ScreenPorch PoolArea PoolQC  Fence MiscFeature  \
0         Lvl    AllPub  ...         120        0    NaN  MnPrv         NaN   
1         Lvl    AllPub  ...           0        0    NaN    NaN        Gar2   
2         Lvl    AllPub  ...           0        0    NaN  MnPrv         NaN   
3         Lvl    AllPub  ...           0        0    NaN    NaN         NaN   
4         HLS    AllPub  ...         144        0    NaN    NaN         NaN   

  MiscVal MoSold  YrSold  SaleType  SaleCondition  
0       0      6    2010        WD         Normal  
1   12500      6    2010        WD         Normal  
2       0      3    2010        WD         Normal  
3       0      6    2010        WD         Normal  
4       0      1    2010        WD         Normal  

[5 rows x 80 columns]

## Data fields

- label column : SalePrice 부동산의 판매 가격 (달러)입니다. 이것이 예측하려는 목표 변수입니다.

- SalePrice- 부동산의 판매 가격 (달러)입니다. 이것이 예측하려는 목표 변수입니다.
- MSSubClass : 건물 등급
- MSZoning : 일반 구역 분류
- LotFrontage : 부동산에 연결된 거리의 선형 피트
- LotArea : 평방 피트 단위의 부지 크기
- Street : 도로 접근 유형
- Alley : 골목 접근 유형
- LotShape : 속성의 일반적인 모양
- LandContour : 부동산의 평탄도
- Utilities : 사용 가능한 유틸리티 유형
- LotConfig : 로트 구성
- LandSlope : 부동산의 경사
- Neighborhood : 에임스시 경계 내의 물리적 위치
- Condition1 : 주요 도로 또는 철도와의 근접성
- Condition2 : 주요 도로 또는 철도와의 근접성 (초가있는 경우)
- BldgType : 주거 형태
- HouseStyle : 주거 스타일
- OverallQual : 전체 재료 및 마감 품질
- OverallCond : 전체 상태 등급
- YearBuilt : 원래 공사 일
- YearRemodAdd : 리모델링 날짜
- RoofStyle : 지붕 유형
- RoofMatl : 지붕 재료
- Exterior1st : 주택 외장재
- Exterior2nd : 주택 외장재 (하나 이상인 경우)
- MasVnrType : 조적 베니어 유형
- MasVnrArea : 벽돌 베니어 면적 (평방 피트)
- ExterQual : 외장재 품질
- ExterCond : 외장재의 현황
- Foundation : 기초 유형
- BsmtQual : 지하 높이
- BsmtCond : 지하실 개황
- BsmtExposure : 워크 아웃 또는 정원 수준의 지하 벽
- BsmtFinType1 : 지하 마감면의 품질
- BsmtFinSF1 : 유형 1 완성 된 평방 피트
- BsmtFinType2 : 두 번째 완성 된 영역의 품질 (있는 경우)
- BsmtFinSF2 : 유형 2 완성 된 평방 피트
- BsmtUnfSF : 미완성 된 지하실 면적
- TotalBsmtSF : 지하 총 평방 피트
- Heating : 난방의 종류
- HeatingQC : 난방 품질 및 상태
- CentralAir : 중앙 에어컨
- Electrical : 전기 시스템
- 1stFlrSF : 1 층 평방 피트
- 2ndFlrSF : 2 층 평방 피트
- LowQualFinSF : 저품질 마감 평방 피트 (모든 층)
- GrLivArea : 지상 (지상) 거실 면적 평방 피트
- BsmtFullBath : 지하 전체 욕실
- BsmtHalfBath : 지하 반 욕실
- FullBath : 등급 이상의 전체 욕실
- HalfBath : 지상 절반의 목욕
- Bedroom : 지하층 이상의 침실 수
- Kitchen : 주방 수
- KitchenQual : 주방 품질
- TotRmsAbvGrd : 학년 이상의 총 방 (화장실 제외)
- Functional : 홈 기능 등급
- Fireplaces : 벽난로 수
- FireplaceQu : 벽난로 품질
- GarageType : 차고 위치
- GarageYrBlt : 차고 건설 연도
- GarageFinish : 차고 내부 마감
- GarageCars : 차량 수용 가능 차고 크기
- GarageArea : 차고의 크기 (평방 피트)
- GarageQual : 차고 품질
- GarageCond : 차고 상태
- PavedDrive : 포장 된 진입로
- WoodDeckSF : 목재 데크 면적 (평방 피트)
- OpenPorchSF : 평방 피트 단위의 오픈 현관 영역
- EnclosedPorch : 닫힌 현관 영역 (평방 피트)
- 3SsnPorch : 평방 피트의 3 계절 현관 면적
- ScreenPorch : 스크린 현관 면적 (평방 피트)
- PoolArea : 수영장 면적 (평방 피트)
- PoolQC : 수영장 품질
- Fence : 울타리 품질
- MiscFeature : 다른 카테고리에서 다루지 않는 기타 기능
- MiscVal : 기타 기능의 $ Value
- MoSold : 월 판매
- YrSold : 판매 연도
- SaleType : 판매 유형
- SaleCondition : 판매 조건

### base performance Calculation

- 우선, 특별히 데이터의 상세한 전 처리를 진행하지않고 주택가격과 유관할것으로 예상되는 feature를 주관적으로 선정하여 fitting후 모델의 베이스성능을 구해본다. 
- 주관적으로 선택한 피쳐는 아래와 같다.

- 'MSSubClass':건물 등급 -> 건물의 품질 또는 가치를 대변하는 지표라고 생각됨.
- 'LotArea' : 평방 피트 단위의 부지 크기 -> 절대적인것은 아니지만, 대개 부지가 클 경우 주택 가격도 비례할것으로 예상됨.
- 'Street' : 도로 접근 유형 -> 해당 주택의 접근성이 뛰어날 경우 가격에 영향을 미칠 수 있을것으로 예상됨.
- 'Alley' : 골목 접근 유형 -> 해당 주택의 접근성이 뛰어날 경우 가격에 영향을 미칠 수 있을것으로 예상됨.
- 'Condition1' : 주요 도로 또는 철도와의 근접성 -> 해당 주택과 이동수단간의 근접성이 뛰어날 경우 가격에 영향을 미칠 수 있을것으로 예상됨.
- 'Condition2' : 주요 도로 또는 철도와의 근접성 (초가있는 경우) -> 해당 주택과 이동수단간의 근접성이 뛰어날 경우 가격에 영향을 미칠 수 있을것으로 예상됨.
- 'BldgType' : 주거 형태 -> 주거형태가 주거 트렌드와 밀접한 관련이 있어, 주택 가격에 영향을 미칠 수 있을것으로 판단함.
- 'HouseStyle' : 주거 스타일 -> 위와 동일
- 'YearRemodAdd' : 리모델링 날짜 -> 리모델링 여부와 시기는 주택가격과 밀접한 관련이 있을것으로 예상됨
- 'GrLivArea' : 지상 (지상) 거실 면적 평방 피트 -> 어떠한 시설이던지 면적과 주택가격은 대체로 비례할것으로 예상됨
- 'BsmtFullBath' : 지하 전체 욕실 -> 어떠한 시설이던지 면적과 주택가격은 대체로 비례할것으로 예상됨
- 'BsmtHalfBath' : 지하 반 욕실 -> 어떠한 시설이던지 면적과 주택가격은 대체로 비례할것으로 예상됨
- 'FullBath' : 등급 이상의 전체 욕실 -> 가치가 높은 욕실을 보유한 주택이 당연히 가격도 높게 형성될 것으로 예상됨.
- 'HalfBath' : 지상 절반의 목욕
- 'KitchenQual' : 주방 품질 -> 가치가 높은 주방을 보유한 주택이 당연히 가격도 높게 형성될 것으로 예상됨.
- 'TotRmsAbvGrd' : 학년 이상의 총 방 (화장실 제외) -> 방의 개수가 많을 수록 면적 및 주택가격도 높아질것으로 예상됨
- 'Functional' : 홈 기능 등급 -> 많은 기능을 보유한 주택의 가격이 높게 형성될것으로 예상됨
- 'Fireplaces' : 벽난로 수 -> 벽난로수는 곧 주택의 가치와 면적에 영향을 줄 수 있고, 이것은 주택가격에도 영향을 미칠 수 있을것으로 예상됨
- 'FireplaceQu' : 벽난로 품질 -> 높은 가치를 지닌 주택의 벽난로가 일반적으로 품질도 더 좋을것으로 예상
- 'PoolArea' : 수영장 면적 (평방 피트) -> 수영장 크기와 주택가격은 대체로 비례할것으로 예상.

## preprocessing

#### 중복 데이터 유무 확인

In [5]:
train.duplicated().sum()

# 중복데이터는 없는것으로 확인

0

model fitting을 위하여 type이 'object'인 아래 컬럼에 대한 preprocessing 필요

- 'Street'
- 'Alley'
- 'Condition1'
- 'Condition2'
- 'BldgType'
- 'HouseStyle'
- 'YearRemodAdd'
- 'GrLivArea'
- 'KitchenQual'
- 'Functional'
- 'FireplaceQu'

### 'Street' column One-Hot_Encoding

In [8]:
print(train.shape)
train['Street'].value_counts()

(1460, 81)


Pave    1454
Grvl       6
Name: Street, dtype: int64

In [9]:
street_list = ['Pave', 'Grvl', 'NaN']
for i in street_list :
    col_name = 'street_'+ i
    train[col_name] = train['Street'] == i

print(train['street_Grvl'].value_counts())
print(train['street_Pave'].value_counts())

False    1454
True        6
Name: street_Grvl, dtype: int64
True     1454
False       6
Name: street_Pave, dtype: int64


In [10]:
train[['Street', 'street_Pave', 'street_Grvl', 'street_NaN']].head()

Street  street_Pave  street_Grvl  street_NaN
0   Pave         True        False       False
1   Pave         True        False       False
2   Pave         True        False       False
3   Pave         True        False       False
4   Pave         True        False       False

### 'Alley' column One-Hot_Encoding

In [11]:
train['Alley'].value_counts()

Grvl    50
Pave    41
Name: Alley, dtype: int64

In [12]:
train['Alley'].isnull().sum()

1369

- 'Alley'컬럼에서 null인 데이터가 총 1,369개이며, 실제 데이터는 91개만 수록되어있다.
- 'Alley'컬럼은 'Type of alley access to property'로 해당 '재산(건축물)에 접근 방식의 특성'을 이야기하는데, value를 보면 'Paved:포장도로', 'Gravel:비포장로', NaN으로 표기된 접근방법이 없는 케이스로 접근방식의 편의성에 따라 주택가격에 영향을 미칠 수 있을 것 같다.
- one-hot-encoding 처리필요. 

In [13]:
alley_list = ['Pave', 'Grvl', 'NaN']
for i in alley_list :
    col_name = "alley_" + i
    train[col_name] = train['Alley'] == i

print(train['alley_Pave'].value_counts())
print(train['alley_Grvl'].value_counts())
print(train['alley_NaN'].value_counts())

False    1419
True       41
Name: alley_Pave, dtype: int64
False    1410
True       50
Name: alley_Grvl, dtype: int64
False    1460
Name: alley_NaN, dtype: int64


In [14]:
train[['Alley', 'alley_Pave', 'alley_Grvl', 'alley_NaN']].head()

Alley  alley_Pave  alley_Grvl  alley_NaN
0   NaN       False       False      False
1   NaN       False       False      False
2   NaN       False       False      False
3   NaN       False       False      False
4   NaN       False       False      False

### 'Condition1' column One-Hot_Encoding

kaggle dec.

'Condition1': Proximity to various conditions("다양한 조건에 대한 근접성")
	
- Artery : Adjacent to arterial street(간선 도로에 인접)
- Feedr : Adjacent to feeder street(피더 스트리트에 인접)
- Norm : Normal	
- RRNn : Within 200' of North-South Railroad(남북 철도 200 피트 이내)
- RRAn : Adjacent to North-South Railroad(남북 철도에 인접)
- PosN : Near positive off-site feature--park, greenbelt, etc.(거의 긍정적 인 오프 사이트 기능 (공원, 그린벨트 등))
- PosA : Adjacent to postive off-site feature(포스트 오프 사이트 기능에 인접)
- RRNe : Within 200' of East-West Railroad(동서 철도 200 피트 이내)
- RRAe : Adjacent to East-West Railroad(동서 철도에 인접)




In [15]:
train.Condition1.value_counts()

Norm      1260
Feedr       81
Artery      48
RRAn        26
PosN        19
RRAe        11
PosA         8
RRNn         5
RRNe         2
Name: Condition1, dtype: int64

In [16]:
train.Condition1.value_counts().index

Index(['Norm', 'Feedr', 'Artery', 'RRAn', 'PosN', 'RRAe', 'PosA', 'RRNn',
       'RRNe'],
      dtype='object')

In [17]:
condition1_list = ['Norm', 'Feedr', 'Artery', 'RRAn', 'PosN', 'RRAe', 'PosA', 'RRNn',
       'RRNe']

for i in condition1_list :
    col_name = 'condition1_' + i
    train[col_name] = train['Condition1'] == i
    
print(train['condition1_Norm'].value_counts())
print(train['condition1_Feedr'].value_counts())
print(train['condition1_Artery'].value_counts())
print(train['condition1_RRAn'].value_counts())
print(train['condition1_PosN'].value_counts())
print(train['condition1_RRAe'].value_counts())
print(train['condition1_PosA'].value_counts())
print(train['condition1_RRNn'].value_counts())
print(train['condition1_RRNe'].value_counts())

True     1260
False     200
Name: condition1_Norm, dtype: int64
False    1379
True       81
Name: condition1_Feedr, dtype: int64
False    1412
True       48
Name: condition1_Artery, dtype: int64
False    1434
True       26
Name: condition1_RRAn, dtype: int64
False    1441
True       19
Name: condition1_PosN, dtype: int64
False    1449
True       11
Name: condition1_RRAe, dtype: int64
False    1452
True        8
Name: condition1_PosA, dtype: int64
False    1455
True        5
Name: condition1_RRNn, dtype: int64
False    1458
True        2
Name: condition1_RRNe, dtype: int64


### 'Condition2' column One-Hot_Encoding

Kaggle dec.

Condition2: Proximity to various conditions (if more than one is present)

- Artery : Adjacent to arterial street(간선 도로에 인접)
- Feedr : Adjacent to feeder street(피더 스트리트에 인접)
- Norm : Normal	
- RRNn : Within 200' of North-South Railroad(남북 철도 200 피트 이내)
- RRAn : Adjacent to North-South Railroad(남북 철도에 인접)
- PosN : Near positive off-site feature--park, greenbelt, etc.(거의 긍정적 인 오프 사이트 기능 (공원, 그린벨트 등))
- PosA : Adjacent to postive off-site feature(포스트 오프 사이트 기능에 인접)
- RRNe : Within 200' of East-West Railroad(동서 철도 200 피트 이내)
- RRAe : Adjacent to East-West Railroad(동서 철도에 인접)

In [18]:
train.Condition2.value_counts()

Norm      1445
Feedr        6
PosN         2
Artery       2
RRNn         2
PosA         1
RRAe         1
RRAn         1
Name: Condition2, dtype: int64

In [21]:
train.Condition2.value_counts().index

Index(['Norm', 'Feedr', 'PosN', 'Artery', 'RRNn', 'PosA', 'RRAe', 'RRAn'], dtype='object')

In [22]:
condition2_list = ['Norm', 'Feedr', 'Artery', 'PosN', 'RRNn', 'PosA', 'RRAn', 'RRAe']

for i in condition2_list :
    col_name = 'condition2_' + i
    train[col_name] = train['Condition2'] == i
    
print(train['condition2_Norm'].value_counts())
print(train['condition2_Feedr'].value_counts())
print(train['condition2_Artery'].value_counts())
print(train['condition2_PosN'].value_counts())
print(train['condition2_RRNn'].value_counts())
print(train['condition2_PosA'].value_counts())
print(train['condition2_RRAn'].value_counts())
print(train['condition2_RRAe'].value_counts())

True     1445
False      15
Name: condition2_Norm, dtype: int64
False    1454
True        6
Name: condition2_Feedr, dtype: int64
False    1458
True        2
Name: condition2_Artery, dtype: int64
False    1458
True        2
Name: condition2_PosN, dtype: int64
False    1458
True        2
Name: condition2_RRNn, dtype: int64
False    1459
True        1
Name: condition2_PosA, dtype: int64
False    1459
True        1
Name: condition2_RRAn, dtype: int64
False    1459
True        1
Name: condition2_RRAe, dtype: int64


### 'BldgType' column One-Hot_Encoding

In [23]:
train['BldgType'].value_counts()

1Fam      1220
TwnhsE     114
Duplex      52
Twnhs       43
2fmCon      31
Name: BldgType, dtype: int64

In [29]:
train['BldgType'].value_counts().sum()

1460

In [24]:
train['BldgType'].value_counts().index

Index(['1Fam', 'TwnhsE', 'Duplex', 'Twnhs', '2fmCon'], dtype='object')

In [30]:
bldgtype_list = ['1Fam', 'TwnhsE', 'Duplex', 'Twnhs', '2fmCon']

for i in bldgtype_list :
    col_name = 'bldgtype_' + i
    train[col_name] = train['BldgType'] == i

print(train['bldgtype_1Fam'].value_counts())
print(train['bldgtype_TwnhsE'].value_counts())
print(train['bldgtype_Duplex'].value_counts())
print(train['bldgtype_Twnhs'].value_counts())
print(train['bldgtype_2fmCon'].value_counts())


True     1220
False     240
Name: bldgtype_1Fam, dtype: int64
False    1346
True      114
Name: bldgtype_TwnhsE, dtype: int64
False    1408
True       52
Name: bldgtype_Duplex, dtype: int64
False    1417
True       43
Name: bldgtype_Twnhs, dtype: int64
False    1429
True       31
Name: bldgtype_2fmCon, dtype: int64


### 'HouseStyle' column One-Hot_Encoding

In [32]:
train['HouseStyle'].value_counts()

1Story    726
2Story    445
1.5Fin    154
SLvl       65
SFoyer     37
1.5Unf     14
2.5Unf     11
2.5Fin      8
Name: HouseStyle, dtype: int64

In [33]:
train['HouseStyle'].value_counts().sum()

1460

In [35]:
train['HouseStyle'].value_counts().index

Index(['1Story', '2Story', '1.5Fin', 'SLvl', 'SFoyer', '1.5Unf', '2.5Unf',
       '2.5Fin'],
      dtype='object')

In [36]:
housestyle_list = ['1Story', '2Story', '1.5Fin', 'SLvl', 'SFoyer', '1.5Unf', '2.5Unf',
       '2.5Fin']

for i in housestyle_list :
    col_name = 'housestyle_' + i
    train[col_name] = train['HouseStyle'] == i

print(train['housestyle_1Story'].value_counts())
print(train['housestyle_2Story'].value_counts())
print(train['housestyle_1.5Fin'].value_counts())
print(train['housestyle_SLvl'].value_counts())
print(train['housestyle_SFoyer'].value_counts())
print(train['housestyle_1.5Unf'].value_counts())
print(train['housestyle_2.5Unf'].value_counts())

False    734
True     726
Name: housestyle_1Story, dtype: int64
False    1015
True      445
Name: housestyle_2Story, dtype: int64
False    1306
True      154
Name: housestyle_1.5Fin, dtype: int64
False    1395
True       65
Name: housestyle_SLvl, dtype: int64
False    1423
True       37
Name: housestyle_SFoyer, dtype: int64
False    1446
True       14
Name: housestyle_1.5Unf, dtype: int64
False    1449
True       11
Name: housestyle_2.5Unf, dtype: int64


### 'YearRemodAdd' column One-Hot_Encoding

In [37]:
train['YearRemodAdd'].value_counts()

1950    178
2006     97
2007     76
2005     73
2004     62
       ... 
2010      6
1983      5
1952      5
1986      5
1951      4
Name: YearRemodAdd, Length: 61, dtype: int64

In [38]:
train['YearRemodAdd'].value_counts().sum()

1460

In [46]:
len(train['YearRemodAdd'].value_counts().index)
# value

61

In [47]:
train['YearRemodAdd'].value_counts().index

Int64Index([1950, 2006, 2007, 2005, 2004, 2000, 2003, 2002, 2008, 1996, 1998,
            1995, 1976, 1999, 1970, 1997, 1977, 2009, 1994, 2001, 1972, 1965,
            1993, 1971, 1959, 1968, 1992, 1978, 1966, 1958, 1990, 1962, 1954,
            1969, 1991, 1963, 1960, 1967, 1980, 1973, 1964, 1989, 1987, 1975,
            1979, 1956, 1953, 1957, 1988, 1955, 1985, 1961, 1981, 1974, 1982,
            1984, 2010, 1983, 1952, 1986, 1951],
           dtype='int64')

# 여기서 부터 다시 진행하자...힘들다...노가다.

#### 일단 주택 가격에 영향을 미칠만한 요소를 주관적으로 선정하여 fitting을 하기위하여 컬럼 전처리를 실시한다.

In [ ]:
label = 'SalePrice'
    
feature = [
'MSSubClass',
'LotArea',
'Street',
'Alley',
'Condition1',
'Condition2',
'BldgType',
'HouseStyle',
'YearRemodAdd',
'GrLivArea',
'BsmtFullBath',
'BsmtHalfBath',
'FullBath',
'HalfBath',
'KitchenQual',
'TotRmsAbvGrd',
'Functional',
'Fireplaces',
'FireplaceQu',
'PoolArea']
feature

In [ ]:
x_train = train[feature]
y_train = train[label]
x_train

## model import

In [ ]:
from lightgbm import LGBMRegressor
from sklearn.model_selection import train_test_split

## train/test data split

In [ ]:
x_train_hp, y_train_hp, x_test_hp, y_test_hp = train_test_split()